### PR TITLE
Add `CreateTopic` method to `MockCluster`

### PR DESCRIPTION
**Pull request by mimikwang**
This commit adds the `CreateTopic` method to `MockCluster`. This allows the creation of topics without having to use a `Producer`.

### CODE REVIEW

File: kafka/mockcluster.go [hunk 1]
```diff
@@ -92,6 +92,18 @@ func (mc *MockCluster) SetRoundtripDuration(brokerID int, duration time.Duration
 	return nil
 }
 
+// CreateTopic creates a topic without having to use a producer
+func (mc *MockCluster) CreateTopic(topic string, partitons, replicationFactor int) error {
```
Comment (jdemeyer) [inline kafka/mockcluster.go:96]: Typo: `partitons`

Comment (mimikwang) [reply kafka/mockcluster.go]: fixed - thank you!

**Comment by milindl** [comment]
Thanks for this PR. Created internal branch and PR #1072 to run CI pipelines on this.

**Comment by milindl** [comment]
Merged with #1072, added CHANGELOG and made sure it passes CI.

**Pull request closed without merging**